extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -41,22 +39,27 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
     rawGeoJson?.let {
       ignoreParsedGeoJsonRegistry[sourceId] = false
       onGeoJsonParsedListenerList.add(onGeoJsonParsed)
-      workerHandler.post {
-        val property = it.toPropertyValue()
-        mainHandler.post {
-          geoJsonParsed = true
-          // we set parsed data when sync setter was not called during background work
-          if (ignoreParsedGeoJsonRegistry[sourceId] == false) {
-            setProperty(property, throwRuntimeException = false)
-            onGeoJsonParsedListenerList.forEach {
-              it.onGeoJsonParsed(this)
-            }
+      scope.launch {
+        val property = getPropertyValue(it)
+        geoJsonParsed = true
+        // we set parsed data when sync setter was not called during background work
+        if (ignoreParsedGeoJsonRegistry[sourceId] == false) {
+          setProperty(property, throwRuntimeException = false)
+          onGeoJsonParsedListenerList.forEach {
+            it.onGeoJsonParsed(this@GeoJsonSource)
           }
         }
       }
-    } ?: run { geoJsonParsed = true }
+    } ?: run {
+      geoJsonParsed = true
+    }
   }
 
+  private suspend fun getPropertyValue(it: GeoJson) =
```

Comment:
can we replace it with extension function for `GeoJson` and have another function name similar to `parseData`.

Reply:
` GeoJson.toPropertyValue()` is an extension function, this function is just a wrap the function call int default dispatcher. IMO, we don't need another extension function have the same functionality with ` GeoJson.toPropertyValue()`